core/cat/rabbit_hole.py

```diff
@@ -1,33 +1,114 @@
 import os
 import time
-import tempfile
+import json
 import mimetypes
 from typing import List, Union
+from urllib.request import urlopen
+from urllib.parse import urlparse
 
 from cat.log import log
 from starlette.datastructures import UploadFile
-from langchain.document_loaders import (
-    PDFMinerLoader,
-    UnstructuredURLLoader,
-    UnstructuredFileLoader,
-    UnstructuredMarkdownLoader,
-)
+from fastapi import HTTPException
 from langchain.docstore.document import Document
+from qdrant_client.http import models
+
+from langchain.document_loaders.parsers import PDFMinerParser
+from langchain.document_loaders.parsers.generic import MimeTypeBasedParser
+from langchain.document_loaders.parsers.txt import TextParser
+from langchain.document_loaders.blob_loaders.schema import Blob
+from langchain.document_loaders.parsers.html.bs4 import BS4HTMLParser
 
 
 class RabbitHole:
     """
 
     """
+
     def __init__(self, cat):
         self.cat = cat
 
+        self.file_handlers = {
+            "application/pdf": PDFMinerParser(),
+            "text/plain": TextParser(),
+            "text/markdown": TextParser(),
+            "text/html": BS4HTMLParser()
+        }
```

Comment:
sorry but here I don't remember how we got along. Let me know if you want this different